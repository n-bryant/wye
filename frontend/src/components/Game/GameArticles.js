import React from "react";
import PropTypes from "prop-types";

import { withStyles } from "@material-ui/core/styles";
import createClassNameHelper from "@n_bryant/classnames-helper";
import JSS_CLASS_NAME_PREFIX from "../../../lib/classNamePrefix";

import styles from "./GameArticles.styles";

/**
 * renders a list of article cards for a game's articles
 */
export const GameArticles = props => {
  const classnames = GameArticles.classnames(props);
  const { data } = props;

  return <div className={classnames.root()}>GameArticles</div>;
};
GameArticles.classnames = createClassNameHelper(
  `${JSS_CLASS_NAME_PREFIX}GameArticles`
);
GameArticles.propTypes = {
  // styles to apply
  classes: PropTypes.shape({
    root: PropTypes.string
  }),
  // article info for a game
  data: PropTypes.array
};
GameArticles.defaultProps = {
  classes: {}
};

export default withStyles(styles)(GameArticles);
