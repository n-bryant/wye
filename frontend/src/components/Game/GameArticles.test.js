import { GameArticles } from "./GameArticles";

describe("GameArticles", () => {
  const props = {
    classes: {
      root: "root",
      link: "link",
      article: "article",
      articleTitle: "articleTitle",
      articleContents: "articleContents",
      subTitle: "subTitle"
    },
    articles: [
      {
        title: "title",
        author: "author",
        printDate: "date"
      }
    ]
  };

  it("should render successfully", () => {
    expect(toJson(shallow(<GameArticles {...props} />))).toMatchSnapshot();
  });
});
