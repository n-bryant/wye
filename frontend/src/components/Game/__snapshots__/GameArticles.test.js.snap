// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameArticles should render successfully 1`] = `
<div
  className="wye-GameArticles root"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={1}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <WithStyles(ForwardRef(Typography))
        gutterBottom={true}
        variant="h1"
      >
        Recent News:
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="0"
      md={6}
      xs={12}
    >
      <Link
        prefetch={false}
      >
        <a
          className="wye-GameArticles__link link"
          target="_blank"
        >
          <div
            className="wye-GameArticles__article article"
          >
            <WithStyles(ForwardRef(Typography))
              className="wye-GameArticles__articleTitle articleTitle"
              gutterBottom={true}
              variant="h2"
            >
              title
            </WithStyles(ForwardRef(Typography))>
            <div
              className="wye-GameArticles__articleContents articleContents"
            >
              <WithStyles(ForwardRef(Typography))
                gutterBottom={true}
                variant="body1"
              >
                <span
                  className="wye-GameArticles__subTitle subTitle"
                >
                  Author:
                   
                </span>
                author
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                variant="body1"
              >
                <span
                  className="wye-GameArticles__subTitle subTitle"
                >
                  Published:
                   
                </span>
                date
              </WithStyles(ForwardRef(Typography))>
            </div>
          </div>
        </a>
      </Link>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;
